content/upcoming/language_migrations.md

```diff
@@ -0,0 +1,522 @@
+
+Now after filtering and quantizing our dataset, we can proceed with building our own transition matrix.
+
+## Minimum-cost flow problem
+
+For every GitHub user, we aggregate annual vectors ; we will call them reactors where each of the 393 elements stands for the amount of bytes coded in the corresponding language and year. After normalization, these reactors resemble  histograms which we need to compare with each other.
+
+Here, an elegant approach which is effective both in coding and computational time, is to use [`PyEMD`](https://pypi.python.org/pypi/pyemd) : a Python wrapper for the [Earth Mover's Distance](https://en.wikipedia.org/wiki/Earth_mover's_distance) which is Numpy friendly. This measure -- better than euclidean distance for histogram comparison -- is particularly beneficial because it is based on [Linear Programming](https://en.wikipedia.org/wiki/Linear_programming) (LP). Indeed, it can be seen as the solution of the following [transportation problem](http://www.me.utexas.edu/~jensen/models/network/net8.html), where \\(~\\sum_ {i=1}^N s_ i = \\sum_ {j=1}^N d_ j = 1\\)
+
+<img src="/post/language_migrations/emd.png" class="fig">
+<p align="center" class="caption">Transportation Problem with supplies and demands</p>
+
+we can see that for every pair of years, the amount of bytes are either considered as "supplies" or "demands". Besides, since \\(~\\sum_ {i=1}^N s_ i = \\sum_ {j=1}^N d_ j = 1\\),
+it becomes a [cost-minimization flow problem](https://en.wikipedia.org/wiki/Minimum-cost_flow_problem). It's formulated below.
+
+\\begin{equation}
+\\arg \\min \\sum_ {i=1}^N \\sum_ {j=1}^N x_ {ij}
+\\end{equation}
+\\begin{equation}\\label{non-negative}
+x_ {ij} \\geq 0,
+\\end{equation}
+\\begin{equation}\\label{sum1}
+\\sum_ {j=1}^N x_ {ij} = s_ i,
+\\end{equation}
+\\begin{equation}\\label{sum2}
+\\sum_ {i=1}^N x_ {ij} = d_ j,
+\\end{equation}
+\\begin{equation}\\label{emd}
+\\sum_ {i=1}^N \\sum_ {j=1}^N x_ {ij} = 1
+\\end{equation}
+
+Let's go a little aside here. First, "costs" of the edges are set equal to 1 in order for us to be unbiased. Second, to reduce our problem to the classical [flow minimization formulation](https://en.wikipedia.org/wiki/Minimum-cost_flow_problem), we have to add an artificial source and sink on both sides of our [bipartite graph](https://en.wikipedia.org/wiki/Bipartite_graph) to ensure flow conservation. However, it is not a crucial point and if you want a brief insight into such a transformation, you can take a look at the last slides of [Stanford's CS97SI](https://web.stanford.edu/class/cs97si/08-network-flow-problems.pdf).
```

Comment:
**this** is not a critical point; the last slides of ... describe that transformation.